builtin/providers/google/resource_compute_autoscaler.go

```diff
@@ -249,22 +287,47 @@ func resourceComputeAutoscalerRead(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	zone := d.Get("zone").(string)
-	scaler, err := config.clientCompute.Autoscalers.Get(
-		project, zone, d.Id()).Do()
+	region, err := getRegion(d, config)
 	if err != nil {
-		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
-			// The resource doesn't exist anymore
-			log.Printf("[WARN] Removing Autoscalar %q because it's gone", d.Get("name").(string))
-			d.SetId("")
-
-			return nil
-		}
+		return err
+	}
 
-		return fmt.Errorf("Error reading Autoscaler: %s", err)
+	zoneList, err := config.clientCompute.Zones.List(project).Do()
+	if err != nil {
+		return err
 	}
 
+	var scaler *compute.Autoscaler
+	for _, zone := range zoneList.Items {
```

Comment:
I'd recommend pulling this logic into a shared utility class. (I'm guessing there will be other zonal infrastructure queried by region).